source/agora/common/Amount.d

```diff
@@ -33,7 +33,7 @@ public struct Amount
     /// Number of units ('cents') per coins
     public static immutable Amount UnitPerCoin   = Amount(10_000_000, true);
     /// Maximum number of BOA coins that can ever be in circulation
-    public static immutable ulong  MaxCoinSupply = 500_000_000;
+    public static immutable ulong  MaxCoinSupply = 4_948_159_353;
```

Comment:
> When `Amount` was created, the goal was simply to ensure we never go over the limit. The idea was to later add a function which would give us the coin supply based on the block height. Do you think that's something we can achieve or it would create too much trouble (e.g. by forcing us to pass extra parameters everywhere) ?

There are around 800+ mentions of Amount(case sensitive/separate word) in our codebase, I think this could be a ticket on its own.